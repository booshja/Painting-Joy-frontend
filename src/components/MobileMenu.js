// dependencies
import React, { useContext } from "react";
import styled from "styled-components";
// components
import { NavLink } from "react-router-dom";
// assets
import Clouds from "../assets/wall_image_complete.jpg";
// context
import MenuContext from "../context/MenuContext";

const StyledMenu = styled.nav`
    background-image: url(${Clouds});
    background-position: center;
    background-repeat: no-repeat;
    background-size: cover;
    z-index: 100;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    height: 100%;
    width: 100%;
    text-align: center;
    position: absolute;
    top: 0;
    left: 0;
    transition: transform 0.3s ease-in-out;
    transform: ${({ open }) => (open ? "translateY(0)" : "translateY(-100%)")};
`;

const StyledLink = styled(NavLink)`
    font-family: "News Cycle", sans-serif;
    font-weight: 400;
    letter-spacing: 2px;
    font-size: 2rem;
    text-transform: uppercase;
    padding: 1rem 0;
    color: #ffffff;
    transition: color 0.3s linear;

    &.active {
        font-weight: 700;
    }
`;

const MenuWrapper = styled.div`
    background-color: rgba(0, 0, 0, 0.5);
    display: flex;
    flex-direction: column;
    justify-content: center;
    padding: 1rem;
`;

const Menu = () => {
    const { menuOpen, setMenuOpen } = useContext(MenuContext);

    return (
        <StyledMenu open={menuOpen}>
            <MenuWrapper>
                <StyledLink exact to="/" onClick={() => setMenuOpen(false)}>
                    Home
                </StyledLink>
                <StyledLink to="/murals" onClick={() => setMenuOpen(false)}>
                    Murals
                </StyledLink>
                <StyledLink to="/more-art" onClick={() => setMenuOpen(false)}>
                    More Art
                </StyledLink>
                <StyledLink to="/contact" onClick={() => setMenuOpen(false)}>
                    Contact Me
                </StyledLink>
                <StyledLink to="/store" onClick={() => setMenuOpen(false)}>
                    Store
                </StyledLink>
            </MenuWrapper>
        </StyledMenu>
    );
};

export default Menu;
