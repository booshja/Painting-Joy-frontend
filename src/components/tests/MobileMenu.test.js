import { render } from "@testing-library/react";
import MobileMenu from "../MobileMenu";
import MenuContext from "../../context/MenuContext";
import CartContext from "../../context/CartContext";
import { BrowserRouter } from "react-router-dom";

/** Smoke Test */
it("renders without crashing, menuOpen true, cart.length > 0", () => {
    render(
        <BrowserRouter>
            <CartContext.Provider value={{ cart: [1, 2, 3] }}>
                <MenuContext.Provider value={{ menuOpen: true }}>
                    <MobileMenu />
                </MenuContext.Provider>
            </CartContext.Provider>
        </BrowserRouter>
    );
});

it("renders without crashing, menuOpen false, cart.length > 0", () => {
    render(
        <BrowserRouter>
            <CartContext.Provider value={{ cart: [1, 2, 3] }}>
                <MenuContext.Provider value={{ menuOpen: false }}>
                    <MobileMenu />
                </MenuContext.Provider>
            </CartContext.Provider>
        </BrowserRouter>
    );
});

it("renders without crashing, menuOpen true, cart.length = 0", () => {
    render(
        <BrowserRouter>
            <CartContext.Provider value={{ cart: [] }}>
                <MenuContext.Provider value={{ menuOpen: true }}>
                    <MobileMenu />
                </MenuContext.Provider>
            </CartContext.Provider>
        </BrowserRouter>
    );
});

it("renders without crashing, menuOpen false, cart.length = 0", () => {
    render(
        <BrowserRouter>
            <CartContext.Provider value={{ cart: [] }}>
                <MenuContext.Provider value={{ menuOpen: false }}>
                    <MobileMenu />
                </MenuContext.Provider>
            </CartContext.Provider>
        </BrowserRouter>
    );
});

/** Snapshot Test */
it("matches snapshot, menuOpen true, cart.length > 0", () => {
    const { asFragment } = render(
        <BrowserRouter>
            <CartContext.Provider value={{ cart: [1, 2, 3] }}>
                <MenuContext.Provider value={{ menuOpen: true }}>
                    <MobileMenu />
                </MenuContext.Provider>
            </CartContext.Provider>
        </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot, menuOpen false, cart.length > 0", () => {
    const { asFragment } = render(
        <BrowserRouter>
            <CartContext.Provider value={{ cart: [1, 2, 3] }}>
                <MenuContext.Provider value={{ menuOpen: false }}>
                    <MobileMenu />
                </MenuContext.Provider>
            </CartContext.Provider>
        </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot, menuOpen true, cart.length = 0", () => {
    const { asFragment } = render(
        <BrowserRouter>
            <CartContext.Provider value={{ cart: [] }}>
                <MenuContext.Provider value={{ menuOpen: true }}>
                    <MobileMenu />
                </MenuContext.Provider>
            </CartContext.Provider>
        </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot, menuOpen false, cart.length = 0", () => {
    const { asFragment } = render(
        <BrowserRouter>
            <CartContext.Provider value={{ cart: [] }}>
                <MenuContext.Provider value={{ menuOpen: false }}>
                    <MobileMenu />
                </MenuContext.Provider>
            </CartContext.Provider>
        </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
});

/** RTL Tests */
it("renders correctly, menuOpen true, cart.length > 0", () => {
    const { getByText } = render(
        <BrowserRouter>
            <CartContext.Provider value={{ cart: [1, 2, 3] }}>
                <MenuContext.Provider value={{ menuOpen: true }}>
                    <MobileMenu />
                </MenuContext.Provider>
            </CartContext.Provider>
        </BrowserRouter>
    );

    getByText("Home");
    getByText("Murals");
    getByText("Contact Me");
    getByText("Store");
    getByText("Cart (3)");
});

it("renders correctly, menuOpen false, cart.length > 0", () => {
    const { getByText } = render(
        <BrowserRouter>
            <CartContext.Provider value={{ cart: [1, 2, 3] }}>
                <MenuContext.Provider value={{ menuOpen: false }}>
                    <MobileMenu />
                </MenuContext.Provider>
            </CartContext.Provider>
        </BrowserRouter>
    );

    getByText("Home");
    getByText("Murals");
    getByText("Contact Me");
    getByText("Store");
    getByText("Cart (3)");
});

it("renders correctly, menuOpen true, cart.length = 0", () => {
    const { getByText } = render(
        <BrowserRouter>
            <CartContext.Provider value={{ cart: [] }}>
                <MenuContext.Provider value={{ menuOpen: true }}>
                    <MobileMenu />
                </MenuContext.Provider>
            </CartContext.Provider>
        </BrowserRouter>
    );

    getByText("Home");
    getByText("Murals");
    getByText("Contact Me");
    getByText("Store");
    getByText("Cart (0)");
});

it("renders correctly, menuOpen false, cart.length = 0", () => {
    const { getByText } = render(
        <BrowserRouter>
            <CartContext.Provider value={{ cart: [] }}>
                <MenuContext.Provider value={{ menuOpen: false }}>
                    <MobileMenu />
                </MenuContext.Provider>
            </CartContext.Provider>
        </BrowserRouter>
    );

    getByText("Home");
    getByText("Murals");
    getByText("Contact Me");
    getByText("Store");
    getByText("Cart (0)");
});
