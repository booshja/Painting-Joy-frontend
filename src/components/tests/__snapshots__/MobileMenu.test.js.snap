// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot, menuOpen false, cart.length = 0 1`] = `
<DocumentFragment>
  <nav
    class="sc-bdnxRM jQbSZm"
  >
    <div
      class="sc-dlnjwi jsQAfO"
    >
      <a
        aria-current="page"
        class="sc-gtsrHT EnKnO active"
        href="/"
      >
        Home
      </a>
      <a
        class="sc-gtsrHT EnKnO"
        href="/murals"
      >
        Murals
      </a>
      <a
        class="sc-gtsrHT EnKnO"
        href="/contact"
      >
        Contact Me
      </a>
      <a
        class="sc-gtsrHT EnKnO"
        href="/store"
      >
        Store
      </a>
      <a
        class="sc-gtsrHT EnKnO"
        href="/cart"
      >
        Cart (0)
      </a>
    </div>
  </nav>
</DocumentFragment>
`;

exports[`matches snapshot, menuOpen false, cart.length > 0 1`] = `
<DocumentFragment>
  <nav
    class="sc-bdnxRM jQbSZm"
  >
    <div
      class="sc-dlnjwi jsQAfO"
    >
      <a
        aria-current="page"
        class="sc-gtsrHT EnKnO active"
        href="/"
      >
        Home
      </a>
      <a
        class="sc-gtsrHT EnKnO"
        href="/murals"
      >
        Murals
      </a>
      <a
        class="sc-gtsrHT EnKnO"
        href="/contact"
      >
        Contact Me
      </a>
      <a
        class="sc-gtsrHT EnKnO"
        href="/store"
      >
        Store
      </a>
      <a
        class="sc-gtsrHT EnKnO"
        href="/cart"
      >
        Cart (3)
      </a>
    </div>
  </nav>
</DocumentFragment>
`;

exports[`matches snapshot, menuOpen true, cart.length = 0 1`] = `
<DocumentFragment>
  <nav
    class="sc-bdnxRM chgfKH"
    open=""
  >
    <div
      class="sc-dlnjwi jsQAfO"
    >
      <a
        aria-current="page"
        class="sc-gtsrHT EnKnO active"
        href="/"
      >
        Home
      </a>
      <a
        class="sc-gtsrHT EnKnO"
        href="/murals"
      >
        Murals
      </a>
      <a
        class="sc-gtsrHT EnKnO"
        href="/contact"
      >
        Contact Me
      </a>
      <a
        class="sc-gtsrHT EnKnO"
        href="/store"
      >
        Store
      </a>
      <a
        class="sc-gtsrHT EnKnO"
        href="/cart"
      >
        Cart (0)
      </a>
    </div>
  </nav>
</DocumentFragment>
`;

exports[`matches snapshot, menuOpen true, cart.length > 0 1`] = `
<DocumentFragment>
  <nav
    class="sc-bdnxRM chgfKH"
    open=""
  >
    <div
      class="sc-dlnjwi jsQAfO"
    >
      <a
        aria-current="page"
        class="sc-gtsrHT EnKnO active"
        href="/"
      >
        Home
      </a>
      <a
        class="sc-gtsrHT EnKnO"
        href="/murals"
      >
        Murals
      </a>
      <a
        class="sc-gtsrHT EnKnO"
        href="/contact"
      >
        Contact Me
      </a>
      <a
        class="sc-gtsrHT EnKnO"
        href="/store"
      >
        Store
      </a>
      <a
        class="sc-gtsrHT EnKnO"
        href="/cart"
      >
        Cart (3)
      </a>
    </div>
  </nav>
</DocumentFragment>
`;
